File: behavioral/command.go
package main

import "fmt"

type Command interface{ Execute() }
type Light struct{}

func (Light) On()  { fmt.Println("Light on") }
func (Light) Off() { fmt.Println("Light off") }

type LightOnCommand struct{ light Light }

func (c LightOnCommand) Execute() { c.light.On() }

type Remote struct{}

func (Remote) Submit(cmd Command) { cmd.Execute() }

func main() {
	light := Light{}
	cmd := LightOnCommand{light}
	Remote{}.Submit(cmd)
}

File: behavioral/command.php
<?php
interface Command { public function execute(): void; }
class Light {
    public function on(){ echo "Light on\n"; }
    public function off(){ echo "Light off\n"; }
}
class LightOnCommand implements Command {
    private Light $light;
    public function __construct(Light $l){ $this->light = $l; }
    public function execute(): void { $this->light->on(); }
}
class Remote {
    public function submit(Command $c){ $c->execute(); }
}

// استفاده
$light = new Light();
$cmd = new LightOnCommand($light);
$remote = new Remote();
$remote->submit($cmd);
